xyz

```diff
@@ -60,27 +65,70 @@ Options:
         Print xyz's version number and exit.
 "
 
-# http://stackoverflow.com/a/246128/312785
-unset CDPATH
-path="${BASH_SOURCE[0]}"
-while [ -h "$path" ] ; do
-  dir="$(cd -P "$(dirname "$path")" && pwd)"
-  path="$(readlink "$path")"
-  [[ $path == /* ]] || path="$dir/$path"
-done
-dir="$(cd -P "$(dirname "$path")" && pwd)"
+inc() {
+  local prerelease_label="$1" increment="$2" version="$3"
+  local number='0|[1-9][0-9]*'
+  local part="$number|[-0-9A-Za-z]+"
+  local pattern="^($number)[.]($number)[.]($number)(-(($part)([.]($part))*))?$"
```

Comment:
This should match the [grammar][1].


[1]: https://github.com/npm/node-semver/blob/master/range.bnf

Reply:
You've missed the `build` part. Not sure if that's really relevant though.

Reply:
> You've missed the `build` part.

That's true. I've never used it. Have you, Aldwin?

Reply:
I've never used it either.

Reply:
Let's omit `build`. I don't understand how it is used, and I'm not aware of xyz users actually using it. If in the future there *is* demand for `build`, we're free to change our minds. :)